src/main/java/com/cloudbees/jenkins/plugins/customtools/CustomTool.java

```diff
@@ -198,8 +198,8 @@ public void check() throws CustomToolException {
      * @since 0.3
      */
     public @Nonnull List<LabelSpecifics> getAppliedSpecifics(@Nonnull Node node) {
-        List<LabelSpecifics> out = new LinkedList<>();
-        if (labelSpecifics != null) {
+        List<LabelSpecifics> out = new LinkedList<LabelSpecifics>();
```

Comment:
this is not a type of refactoring which I would recommend. Implicit definitions help to keep the code shorter here, and there is no typization error risk